controller/tests/array_action/svc/array_mediator_svc_test.py

```diff
@@ -406,20 +411,43 @@ def test_get_object_by_id_volume_success(self):
         volume = self.svc.get_object_by_id("volume_id", "volume")
         self.assertEqual(volume.name, "volume_id")
 
-    def _prepare_mocks_for_create_snapshot(self, support_deduplicated_copy=True, source_has_deduplicated_copy=False):
+    def _get_custom_dedup_cli_volume(self, support_deduplicated_copy, with_deduplicated_copy, name='source_volume',
+                                     pool_name='pool_name'):
+        volume = self._get_cli_volume(with_deduplicated_copy, name=name, pool_name=pool_name)
+        if not support_deduplicated_copy:
+            del volume.deduplicated_copy
+        return volume
+
+    def _prepare_mocks_for_create_snapshot(self, support_deduplicated_copy=True, source_has_deduplicated_copy=False,
+                                           different_pool_site=False):
         self.svc.client.svctask.mkvolume.return_value = Mock()
         self.svc.client.svctask.mkfcmap.return_value = Mock()
-        source_vol_to_copy_from = self._get_cli_volume(source_has_deduplicated_copy)
-        if not support_deduplicated_copy:
-            del source_vol_to_copy_from.deduplicated_copy
+        source_vol_to_copy_from = self._get_custom_dedup_cli_volume(support_deduplicated_copy,
+                                                                    source_has_deduplicated_copy)
+        vols_to_return = [source_vol_to_copy_from, source_vol_to_copy_from]
+
+        if different_pool_site:
+            pools_to_return = [Munch({'site_name': 'pool_site'}),
+                               Munch({'site_name': 'source_volume_site'}),
+                               Munch({'site_name': 'other_volume_site'}),
+                               Munch({'site_name': 'pool_site'})]
+            self.svc.client.svcinfo.lsmdiskgrp.side_effect = self._mock_cli_objects(pools_to_return)
+
+            aux_vols = [self._get_cli_volume(name='other_volume', pool_name='other_volume_pool'),
```

Comment:
please dont abbreviate, `auxiliary_volumes`. (and any other `vol` in the method)

Reply:
done

Reply:
tnx